test/e2e/serial/config/config.go

```diff
@@ -67,3 +76,64 @@ func GetRteCiImage() string {
 	}
 	return nropTestCIImage
 }
+
+func CheckNodesTopology(ctx context.Context) error {
+	kconfigs, err := validator.GetKubeletConfigurationsFromTASEnabledNodes(ctx, e2eclient.Client)
+	if err != nil {
+		return fmt.Errorf("error while trying to get KubeletConfigurations. error: %w", err)
+	}
+
+	var nrtList nrtv1alpha2.NodeResourceTopologyList
+
+	err = e2eclient.Client.List(context.TODO(), &nrtList)
+	if err != nil {
+		return fmt.Errorf("error while trying to get NodeResourceTopology objects. error: %w", err)
+	}
+
+	policies := []nrtv1alpha2.TopologyManagerPolicy{
+		nrtv1alpha2.SingleNUMANodeContainerLevel,
+		nrtv1alpha2.SingleNUMANodePodLevel,
+	}
+	nrts := e2enrt.FilterByPolicies(nrtList.Items, policies)
+	singleNumaNodeNumber := len(nrts)
+	if singleNumaNodeNumber < minNumberOfNodesWithSameTopology {
+		return fmt.Errorf("not enough nodes with valid policy - found %d, need at least %d", len(nrts), minNumberOfNodesWithSameTopology)
+	}
+
+	nrtsSingleNumaNodePodLevel := e2enrt.FilterTopologyManagerPolicy(nrts, nrtv1alpha2.SingleNUMANodePodLevel)
+	podLevelNumber := len(nrtsSingleNumaNodePodLevel)
+
+	nrtsSingleNumaNodeContainerLevel := e2enrt.FilterTopologyManagerPolicy(nrts, nrtv1alpha2.SingleNUMANodeContainerLevel)
+	containerLevelNumber := len(nrtsSingleNumaNodePodLevel)
+
+	if podLevelNumber < minNumberOfNodesWithSameTopology && containerLevelNumber < minNumberOfNodesWithSameTopology {
```

Comment:
this will 100% be false in our CI envs because either we have 2 nodes with container scope OR two with pod scope. What I would think of is to take one thing as the source of truth, say kubeletconfig config, and based the checks on it. so for example, we should expect numberOfNRONodes = (number Of Nodes With Policy Level as configured in kubeletconfig).

Reply:
since the whole point of this PR is adding a initial check verifying NRT is consistent with kubeletconfig, the source of truth should probably be kubeletconfig indeed

Reply:
Sure but, the whole point is to check that both sources are coherent ... and even if the NRT data seems "ok-ish" here and this check does not fail, it would be checked around line 113 that each of the NRT node data is consistent with its Kubeletconfig

Reply:
I have changed the approach in the last commit.

Thanks for the comments .